src/index.js

```diff
@@ -0,0 +1,84 @@
+const fs = () => window[window._fs_namespace];
+let didInit = false;
+
+const ensureSnippetLoaded = () => {
+  const snippetLoaded = !!fs();
+  if (!snippetLoaded) {
+    throw Error('FullStory is not loaded, please ensure the FullStory snippet is executed before calling FullStory API functions');
+  }
+  return true;
```

Comment:
Is it necessary to return anything from this function? We could return a boolean value indicating whether it has loaded, or we could continue to just throw an  exception if it has not loaded. Mixing the two makes for a strange API.

Reply:
It's a short-cut used in this function:

```
const hasFullStoryWithFunction = (...testNames) => {
  const functionsCreated = () => testNames.every(current => fs()[current]);
  return ensureSnippetLoaded() && functionsCreated();
};
```

where `ensureSnippetLoaded()` guards against invoking a function on an undefined FS object. I'll look into a refactor.

Reply:
You still don't need both because `ensureSnippetLoaded` will never return `false`. You could refactor like this:
```js
const hasFullStoryWithFunction = (...testNames) => {
  ensureSnippetLoaded()
  return testNames.every(current => fs()[current]);
};
```

Reply:
That's exactly what I did :)

Reply:
You know what they say about great minds 😄